serpentTools/tests/test_ResultsReader.py

```diff
@@ -0,0 +1,333 @@
+        2. test_metadata:
+                        Check that metadata variables and
+                        their values are properly stored
+            Raises SerpentToolsException
+        3. test_resdata:
+                        Check that time-dependent results variables
+                        and their values are properly stored
+            Raises SerpentToolsException
+        4. test_univdata:
+                        Check that expected states are read
+                        i.e., ('univ', bu, buIdx, days)
+                        For a single state, check that
+                        infExp keys and values are stored.
+                        Check that infUnc and metadata are properly stored
+            Raises SerpentToolsException
+    """
+
+    def test_varsMatchSettings(self):
+        """Verify that the obtained variables match the settings."""
+        self.assertSetEqual(self.expVarSettings, self.reader.settings['variables'])
+
+    def test_metadata(self):
+        """Verify that user-defined metadata is properly stored."""
+        expectedKeys = set(self.expectedMetadata)
+        actualKeys = set(self.reader.metadata.keys())
+        self.assertSetEqual(expectedKeys, actualKeys)
+        for key, expectedValue in six.iteritems(self.expectedMetadata):
+            if isinstance(expectedValue, str):
+                self.assertSetEqual(set(self.reader.metadata[key]),
+                                       set(expectedValue))
+            else:
+                numpy.testing.assert_equal(self.reader.metadata[key],
+                                       expectedValue)
+
+    def test_resdata(self):
+        """Verify that user-defined metadata is properly stored."""
+        expectedKeys = set(self.expectedResdata)
+        actualKeys = set(self.reader.resdata.keys())
+        self.assertSetEqual(expectedKeys, actualKeys)
+        numpy.testing.assert_equal(self.reader.resdata['absKeff'],
+                                   self.expectedKeff)
+        try:
+            numpy.testing.assert_equal(self.reader.resdata['burnDays'],
+                                   self.expectedDays)
+        except:
+            numpy.testing.assert_equal([], self.expectedDays)
+
+    def test_univdata(self):
+        """Verify that results for all the states ('univ', bu, buIdx, days) exist.
+            Verify that the containers for each state are properly created
+            and that the proper information is stored, e.g. infExp keys and values"""
+        expSt0 = self.expectedStates[0]
+        actualStates = set(self.reader.univdata.keys())
+        self.assertSetEqual(set(self.expectedStates), actualStates)   # check that all states are read
+        self.assertSetEqual(set(self.reader.univdata[expSt0].infExp.keys()),
+                            set(self.expectedInfExp))
+        self.assertSetEqual(set(self.reader.univdata[expSt0].metadata.keys()),
+                            set(self.expectedMetaData))
```

Comment:
This is really confusing and I wonder how this worked previously. 
The following change was made:
expectedMetaData was changed to self.expectedUnivMetaData

Reply:
I think because you had two different variables, `expectedMetadata` for the file metadata like version, and `expectedMetaData` for homogenized universe metadata, like number of groups and kinf